### PR TITLE
Aurora

### PR DESCRIPTION
**Pull request by pk0331**
Aurora serverless Provisioned model terraform module

### CODE REVIEW

File: modules/aws/aurora/examples/aurora-mysql/terraform.defaults [hunk 1]
```diff
@@ -0,0 +1,33 @@
+default_tags = {
+    "argonaut.dev/name"        = "test"
+    "argonaut.dev/type"        = "Aurora"
+    "argonaut.dev/manager"     = "argonaut.dev"
+    "argonaut.dev/rds-engine"  = "aurora-mysql"
+    "argonaut.dev/env/dev" = "true"
```
Comment (suryaoruganti) [inline modules/aws/aurora/examples/aurora-mysql/terraform.defaults:6]: Change format to `"argonaut.dev/env" = "dev"`
Also add another line `"env" = "dev"` # this is a standard tag that many 3p tools read and use

@PrashantRaj18198 we need to do this for all the other pieces as well. Bad decision with the wrong format earlier - sorry 😞

File: modules/aws/aurora/variables.tf [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+variable "master_password" {
+  description = "Password for the master DB user. Note - when specifying a value here, 'create_random_password' should be set to `false`"
+  type        = string
+  sensitive   = true
+}
+
+variable "backup_retention_period" {
+  description = "The days to retain backups for. Default `7`"
+  type        = number
+  default     = 7
```
Comment (suryaoruganti) [inline modules/aws/aurora/variables.tf:51]: make this 9

Comment (suryaoruganti) [reply modules/aws/aurora/variables.tf:51]: This needs to be changed to 9 for the other RDS modules as well

File: modules/aws/aurora/variables.tf [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    database_subnets          = list(string)
+    default_security_group_id = string
+    vpc_cidr_block            = string
+  })
+}
+
+variable "apply_immediately" {
+  description = "Specifies whether any cluster modifications are applied immediately, or during the next maintenance window. Default is `false`"
+  type        = bool
+  default     = false
```
Comment (suryaoruganti) [inline modules/aws/aurora/variables.tf:84]: this should be true

File: modules/aws/aurora/variables.tf [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+variable "skip_final_snapshot" {
+  description = "Determines whether a final snapshot is created before the cluster is deleted. If true is specified, no snapshot is created"
+  type        = bool
+  default     = true
```
Comment (suryaoruganti) [inline modules/aws/aurora/variables.tf:90]: this should be false

File: modules/aws/aurora/variables.tf [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+variable "copy_tags_to_snapshot" {
+  description = "Copy all Cluster `tags` to snapshots"
+  type        = bool
+  default     = true
+}
+
+variable "monitoring_interval" {
+  description = "The interval, in seconds, between points when Enhanced Monitoring metrics are collected for instances. Set to `0` to disble. Default is `0`"
+  type        = number
+  default     = 0
```
Comment (suryaoruganti) [inline modules/aws/aurora/variables.tf:108]: let's enable this and keep it to 60

File: modules/aws/aurora/providers.tf [hunk 6]
```diff
@@ -0,0 +1,17 @@
+terraform {
+  required_version = ">= 0.15"
```
Comment (suryaoruganti) [inline modules/aws/aurora/providers.tf:2]: require version >= 1.0

File: modules/aws/aurora/outputs.tf [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+output "cluster_master_password" {
+  description = "The database master password"
+  value       = module.aurora.cluster_master_password
+  sensitive   = true
+}
+
+output "cluster_master_username" {
+  description = "The database master username"
+  value       = module.aurora.cluster_master_username
+  sensitive   = true
```
Comment (suryaoruganti) [inline modules/aws/aurora/outputs.tf:30]: this is not sensitive

**Review by suryaoruganti — CHANGES_REQUESTED** [review]

**Comment by suryaoruganti** [comment]
@pk0331 we also need the list of supported instance types (with arch, cpu and memory) per region so we can have the drop down in the UI.

File: modules/aws/aurora/variables.tf [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+  description = "The days to retain backups for. Default `7`"
+  type        = number
+  default     = 7
+}
+
+# aws_db_subnet_group
+variable "create_db_subnet_group" {
+  description = "Determines whether to create the database subnet group or use existing"
+  type        = bool
+  default     = true
```
Comment (suryaoruganti) [inline modules/aws/aurora/variables.tf:58]: should be false and we should reuse the subnets in the vpc created for the environment

File: modules/aws/aurora/main.tf [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  master_username         = var.master_username
+  create_random_password  = false ##Setting to flase as we want to pass password as input. 
+  master_password         = var.master_password
+  deletion_protection     = var.deletion_protection
+  backup_retention_period = var.backup_retention_period
+  create_db_subnet_group  = var.create_db_subnet_group
+  vpc_id                  = var.vpc.vpc_id
+  subnets                 = var.vpc.database_subnets
+  
+  create_security_group = true
```
Comment (suryaoruganti) [inline modules/aws/aurora/main.tf:28]: have the sg also as a part of the outputs

File: modules/aws/aurora/variables.tf (hunk 8 above)
Comment (accern-pavan) [reply modules/aws/aurora/variables.tf:58]: We are using exisiting subnetes created by VPC Module, Here in Aurora/RDS we reuse them and created a subnet group. This is what we pass it to RDS. RDS takes subnetgroup as input not directly subnetids

**Comment by suryaoruganti** [comment]
@PrashantRaj18198 plz review

**Comment by prajjwaldimri** [comment]
@suryaoruganti Need your opinion on the following fields for Aurora. Do we need to expose these on the UI?

Property name (Possible Behavior)
- Autoscaling related fields (Always enabled)
- DB Parameter Group Name (Randomly generated)
- DB Parameter Cluster Group Name (Randomly generated)
- DB Parameter Group Family (Can be determined based on the engine and its version selected by the user)

**Comment by suryaoruganti** [comment]
Ones that need to be exposed are checked
- [ x] Autoscaling related fields (Always enabled)
- [ ] Backup Retention Period (Default value)
- [ ] DB Parameter Group Name (Randomly generated)
- [ ] DB Parameter Cluster Group Name (Randomly generated)
- [ ] DB Parameter Group Family (Can be determined based on the engine and its version selected by the user)

**Comment by prajjwaldimri** [comment]
@suryaoruganti Can we merge this?

**Review by suryaoruganti — APPROVED** [review]

File: modules/aws/aurora/examples/aurora-mysql/terraform.defaults (hunk 1 above)
Comment (shubhamgrg04) [reply modules/aws/aurora/examples/aurora-mysql/terraform.defaults]: @suryaoruganti  Have added a separate ticket so it doesn't get skipped - https://linear.app/argonaut/issue/ARG-937/refactor-default-tags-for-all-existing-tf-modules

**Pull request merged**